apps/passport-server/src/database/queries/telegram/fetchTelegramEvent.ts

```diff
@@ -340,3 +341,35 @@ export async function fetchTelegramAnonMessagesWithTopicByNullifier(
   );
   return result.rows;
 }
+
+export async function fetchUserBySemaphoreId(
+  client: Pool,
+  semaphoreId: string
+): Promise<TelegramConversation> {
+  const result = await sqlQuery(
+    client,
+    `
+    SELECT *
+    FROM telegram_bot_conversations 
+    WHERE semaphore_id = $1
+    `,
+    [semaphoreId]
+  );
+  return result.rows[0] ?? null;
```

Comment:
Is there ever a legitimate case where there's more than one entry for a user in this table?

Reply:
Yes, a (semaphore_id, telegram_user_id) tuple can be associated with multiple chats. However, we only need the first instance of said tuple, if it exists. 

And the table constraints ensure that you cannot have a semaphore id be associated with multiple telegram_user_ids